src/Widgets/SettingsPopover.vala

```diff
@@ -157,54 +158,63 @@ public sealed class Terminal.SettingsPopover : Gtk.Popover {
                 update_active_colorbutton (dark_button, s.get_string (n));
             }
         });
-
-        show.connect (get_child ().show_all);
     }
 
-    private Gtk.RadioButton add_theme_button (string theme, out Gtk.CssProvider css_provider = null) {
-        var button = new Gtk.RadioButton (null) {
-            action_target = new Variant.string (theme),
+    private Gtk.CheckButton add_theme_button (string theme, out Gtk.CssProvider css_provider = null) {
+        var button = new Gtk.CheckButton () {
             halign = Gtk.Align.CENTER
         };
 
+        button.set_data<string> ("theme", theme);
         css_provider = new Gtk.CssProvider ();
 
-        unowned var style_context = button.get_style_context ();
-        style_context.add_class (Granite.STYLE_CLASS_COLOR_BUTTON);
-        style_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+        button.add_css_class (Granite.STYLE_CLASS_COLOR_BUTTON);
+        button.add_css_class ("theme-button");
+
+        Gtk.StyleContext.add_provider_for_display (
```

Comment:
If you want to add the provider to the display then you also need to give the button a unique css class and make sure to add that class to the selector in the custom CSS as well. Otherwise all buttons are sharing this style.

Maybe better to leave that for a follow up branch since it's just deprecated behavior and won't block a port?